### PR TITLE
Switch to NcSelect and `@nextcloud/vue` 8

### PR DESCRIPTION
**Pull request by juliushaertl**
- [x] Check other multiselect usages
- [x] Difference to the old component
  - [x] No checkmark next to already selected entries in the list
  - [x] Clicking an already selected item in the list doesn't remove it
- [x] Move dialogs to new API and vue components 

* Resolves: # 
* Target version: main

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: src/components/SettingsSelectGroup.vue [hunk 1]
```diff
@@ -21,32 +21,30 @@
   -->
 
 <template>
-	<NcMultiselect v-model="inputValObjects"
+	<NcSelect v-model="inputValObjects"
 		:options="groupsArray"
-		:auto-limit="false"
+		:input-id="uuid"
```
Comment (luka-nextcloud) [inline src/components/SettingsSelectGroup.vue:26]: `uuid` has not been declared in `data()`

File: src/components/SettingsSelectTag.vue
Comment (luka-nextcloud) [inline src/components/SettingsSelectTag.vue:1]: Missing `NcMultiselect` replacement?

File: src/view/DocumentTargetPicker.vue [hunk 2]
```diff
@@ -106,25 +106,20 @@ export default {
...
-				false, // multiselect
-				OC.getCapabilities().richdocuments.mimetypes, // mime filter
-				false, // modal
-				FilePickerType.Choose, // type
-				'',
-				{
-					target: this.$refs.picker,
-				},
-			)
+			getFilePickerBuilder(t('files', 'Select file or folder to link to'))
```
Comment (luka-nextcloud) [inline src/view/DocumentTargetPicker.vue:109]: Missing `await` since `pick()` (at line 122) returns a Promise.

**Review by juliushaertl — COMMENTED** [review]
Thanks @luka-nextcloud Good catches, I've pushed the changes

File: src/components/SettingsSelectTag.vue
Comment (juliushaertl) [reply src/components/SettingsSelectTag.vue:1]: Yes, indeeed actually the component is unused so I dropped it now in favour of the upstream NcSelectTags

**Review by luka-nextcloud — APPROVED** [review]

**Pull request merged**